modules/lang-mustache/src/main/java/org/elasticsearch/script/mustache/TransportMultiSearchTemplateAction.java

```diff
@@ -76,6 +81,9 @@ protected void doExecute(Task task, MultiSearchTemplateRequest request, ActionLi
                 searchRequest = convert(searchTemplateRequest, searchTemplateResponse, scriptService, xContentRegistry, searchUsageHolder);
             } catch (Exception e) {
                 items[i] = new MultiSearchTemplateResponse.Item(null, e);
+                if (ExceptionsHelper.status(e).getStatus() > 500 && ExceptionsHelper.isNodeOrShardUnavailableTypeException(e) == false) {
+                    logger.warn("MultiSearchTemplate convert failure for request " + request, e);
```

Comment:
I looked at what "request#toString" gives us in terms of details here and there doesn't seem to be much else than the class name (which should be the same for all items) and the parentTask which at least gives us some id. I'm wondering if there is anything else useful here in terms of debugging information, i.e. I'm not sure how useful it would be to get the index into the items array here or if that is not useful.

Reply:
Good catch. I changed it to render the XContent as JSON and that looks pretty good.

Example log line from manual testing:

```
MultiSearchTemplate convert failure for request {"id":"my-search-template","params":{"size":10,"from":0,"query_string":"hello world"},"explain":false,"profile":false}
```